20-red-knob-sea-star/api_definitions/rest/claims_repository.yaml

```diff
@@ -0,0 +1,367 @@
+      description: Endpoint used to store a new claim about a user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal ID of the user
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ClaimData'
+        required: true
+      responses:
+        '201':
+          description: Claim was successfully stored.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Claim'
+        '400':
+          description: Claim cannot be stored.
+        '409':
+          description: Claim was already stored.
+        '422':
+          description: Validation error in submitted ID or claims data.
+
+    delete:
+      operationId: delete_claims
+      summary: Delete all claims about a given user
+      description: Endpoint used to delete all claims about a given user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal user ID
+            type: string
+      responses:
+        '204':
+          description: User claims were successfully deleted.
+        '404':
+          description: The user was not found.
+        '422':
+          description: Validation error in submitted user ID.
+
+  /users/{user_id}/claims/{claim_id}:
+
+    delete:
+      operationId: delete_claim
+      summary: Delete an existing user claim
+      description: Endpoint used to delete an existing user claim.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal user ID
+            type: string
+        - name: claim_id
+          in: path
+          required: true
+          schema:
+            title: Internal claim ID
+            type: string
+      responses:
+        '204':
+          description: User claim was successfully deleted.
+        '404':
+          description: The user claim was not found.
+        '422':
+          description: Validation error in submitted user or claim ID.
+
+    patch:
+      operationId: patch_user
+      summary: Revoke an existing user claim
+      description: Endpoint used to revoke a claim for a specified user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal user ID
+            type: string
+        - name: claim_id
+          in: path
+          required: true
+          schema:
+            title: Internal claim ID
+            type: string
+        - name: revocation_date
+          in: query
+          required: true
+          schema:
+            title: Revocation Date
+            type: string
```

Comment:
This should go into the body as it is not used for resource identification but is the data to patch.

Reply:
Yes, that was meant to go into the body. Changed in d0197ce5d142b3241abf4af5aa7e9d94d7946d66.